projects/core/src/checkout/facade/payment-type.service.ts

```diff
@@ -0,0 +1,52 @@
+import { Injectable } from '@angular/core';
+import { select, Store } from '@ngrx/store';
+import { Observable } from 'rxjs';
+import { pluck, shareReplay, tap, withLatestFrom } from 'rxjs/operators';
+import { PaymentType } from '../../model/cart.model';
+import { StateWithProcess } from '../../process/store/process-state';
+import { getProcessStateFactory } from '../../process/store/selectors/process-group.selectors';
+import { CheckoutActions } from '../store/actions/index';
+import {
+  GET_PAYMENT_TYPES_PROCESS_ID,
+  StateWithCheckout,
+} from '../store/checkout-state';
+import { CheckoutSelectors } from '../store/selectors/index';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class PaymentTypeService {
+  constructor(
+    protected checkoutStore: Store<StateWithCheckout | StateWithProcess<void>>
+  ) {}
+
+  /**
+   * Get payment types
+   */
+  getPaymentTypes(): Observable<PaymentType[]> {
+    return this.checkoutStore.pipe(
+      select(CheckoutSelectors.getAllPaymentTypes),
+      withLatestFrom(
+        this.checkoutStore.pipe(
+          select(getProcessStateFactory(GET_PAYMENT_TYPES_PROCESS_ID))
```

Comment:
This is more of an overall design question that spans many files ( actions / reducer / service, etc ).  

For loading data and manage loading state we can use the   loaderReducer and have action extend StateUtils.LoaderLoadAction and such.

Since we load a list of payment types, I kind of expected something similar.  I'd like to understand why we need to load "entities" and why we can't use the loaderReducer.  Let's discuss.

Reply:
From my understanding, if the data is a map, then we want it to be `entities`. For `payment types` data, it will be changed when language changed. So, `getPaymentTypes` function check `loadingState`. And, we also have function `setPaymentType`, when this function is called, the state is `loading`, so that the data is not reloaded during this call. You can take a look at "SupportedDeliveryModes". This PaymentTypes does the same thing as it.

Reply:
Thanks for the explanation.